msc_pygeoapi/process/weather/extract_sounding_data.py

```diff
@@ -0,0 +1,640 @@
+
+
+@click.group('execute')
+def extract_sounding_data_execute():
+    pass
+
+
+@click.command('extract-sounding-data')
+@click.pass_context
+@click.option('--model', help='GDPS, RDPS or HRDPS', required=True)
+@click.option(
+    '-mr',
+    '--model_run',
+    help='model run in %Y-%m-%dT%H:%M:%SZ format',
+    required=True,
+)
+@click.option(
+    '-fh',
+    '--forecast_hour',
+    help='forecast hour 3 digits format',
+    required=True,
+)
+@click.option(
+    '--lon',
+    help='longitude in number format, i.e. -85.000, not 85.000W',
+    required=True,
+)
+@click.option(
+    '--lat',
+    help='latitude in number format, i.e. -85.000, not 85.000S',
+    required=True,
+)
+@click.option(
+    '-c',
+    '--convection_indices',
+    is_flag=True,
+    help=(
+        'True to return convection indices, '
+        'False otherwise. Defaults to True. '
+    ),
+    required=False,
+    default=True,
+)
+@click.option(
+    '-t',
+    '--temperature_data',
+    is_flag=True,
+    help=(
+        'True to return temperature and dewpoint sounding data, '
+        'False otherwise. Defaults to True. '
+    ),
+    required=False,
+    default=True,
+)
+@click.option(
+    '-w',
+    '--wind_data',
+    is_flag=True,
+    help=(
+        'True to return wind sounding data, '
+        'False otherwise. Defaults to True. '
+    ),
+    required=False,
+    default=True,
+)
+@click.option(
+    '-n',
+    '--novalues_above_100mbar',
+    is_flag=True,
+    help='Set flag if you don\'t need values for pressure levels above 100mb',
+    required=False,
+)
+def extract_sounding_data_cli(
+    ctx,
+    model,
+    model_run,
+    forecast_hour,
+    lon,
+    lat,
+    convection_indices,
+    temperature_data,
+    wind_data,
+    novalues_above_100mbar,
+):
+    import rasterio
+    import time
+
+    start = time.time()
+    with rasterio.Env():
+        output = extract_sounding_data(
+            model,
+            model_run,
+            forecast_hour,
+            float(lon),
+            float(lat),
+            convection_indices,
+            temperature_data,
+            wind_data,
+            novalues_above_100mbar,
+        )
+    end = time.time()
+    print('Time elapsed array:', end - start)
+    click.echo(json.dumps(output, ensure_ascii=False))
+
+
+extract_sounding_data_execute.add_command(extract_sounding_data_cli)
+
+try:
+    from pygeoapi.process.base import BaseProcessor, ProcessorExecuteError
+
+    class ExtractSoundingDataProcessor(BaseProcessor):
+        """Extract Sounding Data Processor"""
+
+        def __init__(self, provider_def):
+            """
+            Initialize object
+            :param provider_def: provider definition
+            :returns: pygeoapi.process.weather.extract_sounding_data.ExtractSoundingDataProcessor  # noqa
+            """
+
+            BaseProcessor.__init__(self, provider_def, PROCESS_METADATA)
+
+        def execute(self, data):
+            mimetype = 'application/json'
+
```

Comment:
Validation: suggest to add a check for required dict keys and throw error if they are not provided:

```python
required = ['model', 'model_run']

if None in required:
    msg = 'missing required parameters'
    LOGGER.error(msg)
    raise ProcessorExecuteError(msg)